function-app/adb-to-purview/src/Function.Domain/Services/BlobClientFactory.cs

```diff
@@ -3,13 +3,18 @@
 using Microsoft.Extensions.Configuration;
 using System;
 using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Text;
+using System.IO;
+using Azure.Storage.Blobs.Models;
 
 namespace Function.Domain.Services
 {
     public class BlobClientFactory : IBlobClientFactory
```

Comment:
@kantmani4 

THe original intent of this Factory is to be a singleton and return ne winstances of BlobClient as needed. It was typed to a certain storage container..

https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-client-management?tabs=dotnet#manage-client-objects

Reading the documentation some more, it seems it's safe to create a singleton of BlobServiceclient only and create multiple instances of BlobContainerClient.

I would then recommend having 1 method in here:

public BlobClient Create(string containerName, string name).

This operation should use the singleton BlobServiceClient _blobServiceClient field, and make a call to GetBlobContainerCLient and GetBlobClient accordingly. We cannot store an instance of GetblobContainerClient b.c this class is a Singleton and issues will come up (race conditions).

Reply:
Done